### PR TITLE
Update DG

### PR DESCRIPTION
**Pull request by Btaykb**
In this PR:

- Add a short Use Case summary, don't exactly know how to include it at the start of the DG as it will look quite out of place, so for now the summary is before the Use Cases
- Add some use cases for the other features added in v1.3
- Add some whitespace between sections for better readability (Use case and Implementation Details section)
- For Gui for adding, editing section, split into steps for easier readability 
- Update value proposition

Some of the reordering of alternatives and changing of pros and cons to a list format is already handled in [here](https://github.com/AY2122S2-CS2103T-T13-4/tp/pull/105).

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -264,18 +270,23 @@ Implement a separate command to handle the removal of `comment`.
 The GUI for `AddWindow` and `EditWindow` are done using JavaFX with SceneBuilder.  
 The adding and editing mechanism is driven by the CLI commands, `add` and `edit`, and both goes through their respective `Parser`
 
-User input is retrieved from their respective `TextField`  
-User input is strung together to follow the proper `Command` format, which is then passed to `Logic` to handle the rest of the execution  
+Step 1: User input is retrieved from their respective `TextField`.
```
Comment (junlee1991) [inline docs/DeveloperGuide.md:273]: instead of 'their', I think it should be its since user input is singular.

**Comment by junlee1991** [comment]
Other than some minor grammar issues, LGTM

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -407,13 +421,16 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
 ### Use cases
 
-(For all use cases below, the **System** is `ModuleMateFinder` and the **Actor** is the `user`, unless specified otherwise)
+ModuleMateFinder provides the necessary features that support the management of contact information such as adding,
+deleting, listing, sorting, finding and editing. Moreover, there are module related features that are more specific
+to the context of ModuleMateFinder. The Use Cases listed below demonstrates their usages.
```
Comment (junlee1991) [inline docs/DeveloperGuide.md:426]: think this should be 'demonstrate'

**Comment by Btaykb** [comment]
> Other than some minor grammar issues, LGTM

Agreed on the grammar issues, will make the changes. Thanks for pointing those out!

**Comment by bakano98** [comment]
Yep, other than the things already mentioned, LGTM.

**Pull request merged**